website/docs/components/code-block/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,100 @@
+
+Use line highlighting to target and call attention to specific lines or multiple lines within a block of code.
+
+![Example of line highlighting in the Code Block](/assets/components/code-block/code-block-line-highlighting.png)
+
+!!! Info
+
+In the Ember component, lines can be highlighted by passing a single line number, multiple line numbers, or a range of lines. For more examples refer to the [How to use](/?tab=code#highlightlines) documentation. 
+
+In the Figma component, line highlighting (`highlightLines`) is a boolean property that displays a _representative_ example of what a highlighted line will look like visually and is not a property that can be controlled by the consumer. Instead, lines that are intended to be highlighted should be communicated in the engineering handoff or annotated in the design file.
+!!!
+
+## Language
+
+Language determines how syntax highlighting is applied and formatted within the block, but is handled a bit differently between the Ember and Figma components.
+
+The **Ember** component uses [Prism.js](https://prismjs.com/index.html) to handle syntax highlighting and supports close to [300 different languages and syntaxes](https://prismjs.com/index.html#supported-languages). The manner in which the syntax highlighting is applied is dependent on the value passed to the component.
+
+In **Figma** we provide a handful of example languages that are intended to be _representative_ of the end result in production. Syntax highlighting in Figma is a non-trivial process and requires the manual application of color styles to each "type" of code. Despite this, creating a custom code snippet with the Code Block is supported by typing/pasting into the text layer, but syntax highlighting will not be automatically applied.
+
+### Applying syntax highlighting
+
+If you wish to create custom examples using the Code Block, we publish all of the relevant syntax highlighting styles in the HDS [Foundations](https://www.figma.com/file/oQsMzMMnynfPWpMEt91OpH/HDS-Product---Foundations?type=design&node-id=2130%3A2&mode=design&t=Pfj7CheLS6cR0hKa-1) library. However, due to the number of languages supported by the component, the color styles use a generic naming schema (e.g., cyan, red, purple) to remain as agnostic as possible when being applied to different languages.
+
+!!! Info
+
+Working directly with an engineering partner can reveal exactly how a snippet will render in the component and should be the first course of action when creating custom snippets. Understanding Prism's [token hierarchy](https://prismjs.com/tokens.html) can also be helpful when creating examples.
+
+If you have questions or need assistance creating custom examples, don’t hesitate to reach out the the HDS team for [support](/about/support).
+!!!
+
+To aid in understanding how the highlighting theme is applied via Prism's tokens, we've provided a high-level, non-exhaustive list of token names and how they might be applied depending on the syntax.
```

Comment:
Maybe, to reduce the content of this section, which is already very dense and technical, this could be moved to the "Specifications" tab?

Reply:
👍 I like that idea

Reply:
Moved this more detailed information to the specs tab